docs/vector/vector-intro.md

```diff
@@ -83,122 +85,87 @@ myML.unionGeometries
 about what was in the head of the author.
-There's only one package which does anything with `D`, so our job
-should be relatively easy. From `geotrellis.vector.io.json.FeatureFormats.scala`:
-
-```Scala
-def writeFeatureJson[D: JsonWriter](obj: Feature[D]): JsValue = {
-  JsObject(
-    "type" -> JsString("Feature"),
-    "geometry" -> GeometryFormat.write(obj.geom),
-    "properties" -> obj.data.toJson
-  )
-}
-def readFeatureJson[D: JsonReader, G <: Geometry: JsonReader, F <: Feature[D]](value: JsValue)(create : (G, D) => F): F = {
-  value.asJsObject.getFields("type", "geometry", "properties") match {
-    case Seq(JsString("Feature"), geom, data) =>
-      val g = geom.convertTo[G]
-      val d = data.convertTo[D]
-      create(g,d)
-    case _ => throw new DeserializationException("Feature expected")
-  }
-}
-```
 
-`D`'s purpose is clear: any `D` which comes with the tools
+There's only one `package` which does anything with `D`, so the
+constraints (and our job) should be relatively easy.
+In [`geotrellis.vector.io.json.FeatureFormats.scala`](../../vector/src/main/scala/vector/io/json/FeatureFormats.scala)
+there are `ContextBound`s on `D` which ensure that they have JsonReader,
+JsonWriter, and JsonFormat implicits available (this is a
+[typeclass](http://danielwestheide.com/blog/2013/02/06/the-neophytes-guide-to-scala-part-12-type-classes.html),
+and it allows for something like type-safe duck-typing).
+
+`D`'s purpose is clear enough: any `D` which comes with the tools
 necessary for json serialization and deserialization will suffice.
 In effect, `data` corresponds to the "properties" member of the
-geojson spec's Feature object.
+geojson spec's `Feature` object.
 
-If you can provide the serialization tools (almost certainly implicit
-conversions between some case class and [spray json](https://github.com/spray/spray-json)),
+If you can provide the serialization tools (that is, implicit
+conversions between some type (your `D`) and [spray json](https://github.com/spray/spray-json)),
 the `Feature` object in `geotrellis.vector` does the heavy lifting
 of embedding your (thus serializable) data into the larger structure
-which includes a geometry. There's even support for geojson IDs: the
+which includes a `Geometry`. There's even support for geojson IDs: the
 "ID" member of a geojson Feature is represented by the keys of a `Map`
 from `String` to `Feature[D]`. Data in both the ID and non-ID variants
 of geojson Feature formats is easily transformed.
 
-##GeoTrellis Extents
+## GeoTrellis Extents
 
 There's one more piece to the `geotrellis.vector` puzzle: `Extent`.
 A `geotrellis.vector.Extent` is nothing more than a rectangular
-polygon which is projected (look
-[here](../../../../../proj4/src/main/scalageotrellis/proj4) for more
-on projection). This is useful mainly as a tool for defining the
+polygon on projection). This is useful mainly as a tool for defining the
```

Comment:
Something wonky here - this closing parenthesis does not correspond to any opening parenthesis.